### DISCUSSION TITLE
`consult-after-jump-hook` but not for preview

### DISCUSSION THREAD

**Discussion by neurolit** [minad/consult#1059] [Q&A] [answered]
Hi!

I heavily use `consult-outline` in order to navigate in very long org files. Moreover, I'm using [dogears](https://github.com/alphapapa/dogears.el) in order to remember the last places my point was: `dogears-remember` is called by hooks in order to add the current place into the history, easy peasy. When I use `consult-imenu`, I can add `dogears-remember` into `imenu-after-jump-hook`, and the place is remembered when I **select** the candidate, and not during preview. However, when I use `consult-outline` (which I prefer), if I add `dogears-remember` to `consult-after-jump-hook`, each preview position is remembered (as intended: "The hook called during preview
and for the jump after selection").

Is there a way to use another hook in order to call `dogears-remember` when the candidate is selected by `consult-outline` but not during the preview (like `imenu-after-jump-hook`) ? Or is it possible to add such a hook?

**Comment by oantolin** [comment, marked as answer]
Can't you just `(advice-add 'consult-outline :after 'dogears-remember)`?

**Reply by neurolit** [reply]
You're totally right, using an advice didn't occur to me :man_facepalming: 

Thank you for your suggestion! My problem should be solved that way.

However, I think there is a small incoherence between `consult-outline` and `consult-imenu`, as the first one calls `consult-after-jump-hook` during preview, whereas the second doesn't with `imenu-after-jump-hook`.

**Reply by neurolit** [reply]
And, of course, thank you very much for [orderless](https://github.com/oantolin/orderless), which I use all day long.
